tokenizer/src/main/java/edu/illinois/cs/cogcomp/nlp/tokenizer/TokenizerStateMachine.java

```diff
@@ -473,21 +473,24 @@ protected void push(State newState, int where) {
      * @param intext the text to parse.
      */
     protected void parseText(String intext) {
-    	// find index of last non whitespace
-    	int i = intext.length()-1;
-    	for (; i >=0 ; i--) {
-    		if (!Character.isWhitespace(intext.charAt(i)))
-    			break; // found the first non-whitepsace.
-    	}
-    	i++; // length is one beyond the last whitespace char.
+    	// CHANGE (MS): preserve trailing whitespace (disabled find index of last non whitespace)
```

Comment:
Is this always the desired option? Or should this be a boolean switch (`preserveWhitespace`)?